### PR TITLE
Update to 0.11

### LINKED ISSUE #64: Remove `--disable-rpc-request-validation` from `DevnetClient.swift` when felt validation in devnet is fixed

### LINKED ISSUE #72: Remove `typeguard` and `Web3` from `requirements.txt` after PR merge in `cairo-lang`

### PR DESCRIPTION
**Pull request by bartekryba**
Updated sdk to support starknet 0.11

Waiting for starknet-devnet to merge rpcv3 update. After that, the tests will pass.

Closes #64, closes #72

To test it locally, use branch https://github.com/bartekryba/starknet-devnet/tree/rpc-v3 of starknet devnet.

### CODE REVIEW

File: Sources/Starknet/Data/Transaction/TransactionWrapper.swift [hunk 1]
```diff
@@ -11,7 +11,8 @@ enum TransactionWrapper: Decodable {
     case invokeV0(StarknetInvokeTransactionV0)
     case deployAccount(StarknetDeployAccountTransaction)
     case deploy(StarknetDeployTransaction)
-    case declare(StarknetDeclareTransaction)
+    case declareLegacy(StarknetDeclareTransactionLegacy)
```
Comment (THenry14) [inline Sources/Starknet/Data/Transaction/TransactionWrapper.swift:14]: Maybe let's use one pattern for all versioned methods, and name it declareV1?

**Review by THenry14 — COMMENTED** [review]
overall looks good
I suppose we won't be adding new methods/things like state_update anytime soon?

File: Sources/Starknet/Data/Transaction/TransactionWrapper.swift (hunk 1 above)
Comment (bartekryba) [reply Sources/Starknet/Data/Transaction/TransactionWrapper.swift:14]: I named this way on purpose, as this declareLagacy can be both v0 and v1

Comment (THenry14) [reply Sources/Starknet/Data/Transaction/TransactionWrapper.swift:14]: I think declare 0 is already deprecated and removed, but let's double check

**Review by THenry14 — APPROVED** [review]
lgtm :+1:

**Pull request merged**